lib/api/src/js/instance.rs

```diff
@@ -19,8 +23,10 @@ use std::fmt;
 #[derive(Clone)]
 pub struct Instance {
     _handle: StoreHandle<WebAssembly::Instance>,
+    thread_seed: Arc<AtomicU32>,
```

Comment:
What is this for?

Reply:
The thread ID is abstracted so that real thread ID's are not leaked into the sandbox (security considerations) - this seed is used to generate incrementing threads - it could be moved into WasiEnv but then threading starts to become a WASI only feature while I think its likely other things will use threading (unless we just use WASI code for none WASI ABI's)

Reply:
this is now moved away from Instance and put into WasiStateThreading instead